src/bgw/job_stat.c

```diff
@@ -161,18 +162,61 @@ typedef struct
 	BgwJob *job;
 } JobResultCtx;
 
+static TimestampTz
+get_next_scheduled_execution_slot(BgwJob *job, TimestampTz finish_time)
+{
+	Assert(job->fd.fixed_schedule == true);
+	Datum timebucket_fini, result;
+	Datum schedint_datum = IntervalPGetDatum(&job->fd.schedule_interval);
+
+	timebucket_fini = DirectFunctionCall3(ts_timestamptz_bucket,
+										  schedint_datum,
+										  TimestampTzGetDatum(finish_time),
+										  TimestampTzGetDatum(job->fd.initial_start));
+	// always the next time_bucket
+	result = DirectFunctionCall2(timestamptz_pl_interval, timebucket_fini, schedint_datum);
+
+	return DatumGetTimestampTz(result);
+}
```

Comment:
This is correct, but is there a reason to why you do not just add the interval to the last scheduled start time?

Reply:
I think I need to do this since the last scheduled start time may have drifted due to failures. If we only have successes, the other option would be to use one of the two fields in `bgw_job_stats`: `next_start` or `last_start`. The next_start gets erased when the job starts, so I would need to make some changes to get it before it is erased. The `last_start` is not exactly aligned with the origin because there is some small delay in launching the job.